internal/compiler/generator/cpp.rs

```diff
@@ -2441,8 +2441,8 @@ fn compile_builtin_function_call(
                 let popup_window_id =
                     ident(&current_sub_component.popup_windows[*popup_index as usize].root.name);
                 let parent_component = access_item_rc(parent_ref, ctx);
-                let x = compile_expression(x, ctx);
-                let y = compile_expression(y, ctx);
+                let x = format!("(float){}", compile_expression(x, ctx));
+                let y = format!("(float){}", compile_expression(y, ctx));
                 format!(
                     "{}.show_popup<{}>({}, {{ {}, {} }}, {{ {} }})",
```

Comment:
I suggest avoiding the C casts if possible.  Using `float(...)` or `static_cast` is preferable.  (In the other change with `/`  it still use a cast because otherwise extra parentheses would be needed where it's not trivial to add them)
Also, since the string construction happens later, we don't need to build intermediate string.  suggestion:

```suggestion
                    "{}.show_popup<{}>({}, {{ static_cast<float>({}), static_cast<float>({}) }}, {{ {} }})",
```
The new syntax for format string could also be used now:

```suggestion
                    "{window}.show_popup<{popup_window_id}>({component_access}, {{ static_cast<float>({x}), static_cast<float>({y}) }}, {{ {parent_component} }})",
```